pysnn/learning.py

```diff
@@ -1,3 +1,6 @@
+from collections import defaultdict
```

Comment:
`defaultdict` is unused? So are `torch.nn`, `pysnn.connection` and `pysnn.utils`